portfolio/src/main/webapp/discussionScript.js

```diff
@@ -13,16 +13,75 @@ function displayList() {
       content.classList.add("post-content");
       content.textContent = data[i].content;
 
+      let replies = prepareReplies(data[i].replies);
+      replies.classList.add("post-replies");
 
       let singlePost = document.createElement('div');
       singlePost.classList.add("post");
 
+      let replyButton = document.createElement('button');
+      replyButton.textContent = "Reply to this post";
+
       singlePost.appendChild(title);
       singlePost.appendChild(content);
+      singlePost.appendChild(replies);
+      singlePost.appendChild(replyButton);
 
       posts.appendChild(singlePost);
+
+      replyButton.addEventListener("click", () => {
+        replyButton.classList.add("hidden");
+        createReplyForm(singlePost, data[i].id);
+      });
     }
 
     document.getElementById('discussion-container').appendChild(posts);
   })
 }
+
+function createReplyForm(singlePost, postId){
+
+  let replyForm = document.createElement("form");
+  replyForm.action = `/reply?postId=${postId}`;
+  replyForm.method = "POST";
+
+  let inputBox = document.createElement("textarea");
+  inputBox.name = "reply-content"
+  inputBox.rows = "2";
+  inputBox.cols = "40";
+
+  let submitButton = document.createElement("input");
+  submitButton.type = "submit";
+  submitButton.value = "Post";
+
+  replyForm.appendChild(inputBox);
+  replyForm.appendChild(submitButton);
+
+  singlePost.appendChild(replyForm);
+}
+
+function prepareReplies(singlePostReplies){
+
+  let allReplyContainer = document.createElement("div");
+
+  for (let i = 0; i < singlePostReplies.length; ++i) {
+
+    let replyTime = new Date(1591370080881).toLocaleString();
```

Comment:
I dont see this being used, delete?

Reply:
Oops. I was just testing if it worked. Deleted it!